src/js/store/middleware/feature/form.js

```diff
@@ -257,6 +336,7 @@ export const formMiddleware = store => next => action => {
 
     case INPUT_FIELD:
       store.dispatch(setField({ fieldId: action.meta.fieldId, value: action.payload }))
+      store.dispatch(persistInLocalStorage())
```

Comment:
An important addition is that we dispatch a `PERSIST_IN_LOCAL_STORAGE` action in response to `INPUT_FIELD` actions (again, that's the action triggered by manual user input). This means that each time the user interacts with the form, we store their work in local storage, so if they leave before submitting the page they can retrieve everything in the future.

Reply:
I think this is where I have my main outstanding question. How frequently does that `INPUT_FIELD` action get triggered? For example, does this dispatch end up happening after every keypress in a text field? (And if so, should we throttle that somehow?)

Reply:
Yeah, that's a good point. This does get triggered on each `onChange` event, so on each keypress in a text field. I haven't seen any latency issues or anything (keyboard mashing on a text input seems totally fine), but we certainly could debounce the actual local storage write. I think I'll go with this as-is for now and open a ticket for that improvement. Thanks for flagging it!